equational_theories/Preorder.lean

```diff
@@ -0,0 +1,36 @@
+import Mathlib.Order.Defs
+import Mathlib.Data.Set.Basic
+import equational_theories.Completeness
+
+namespace MagmaLaw
+
+variable {α : Type}
+
+/--
+A magma law `l₁` implies a law `l₂` if in any Magma `G` where `l₁` holds, `l₂` also holds.
+
+We have to explicitly quantify the type `G` and the Magma instance `[Magma G]` instead of
+using them as parameters so that the implication holds in any Magma `G`.
+-/
+def implies (l₁ l₂ : MagmaLaw α) := ∀ {G : Type} [Magma G],
+  satisfies G l₁ → satisfies G l₂
+
+instance : LE (MagmaLaw α) where
+  le l₁ l₂ := l₁.implies l₂
+
+theorem implies_refl (l : MagmaLaw α) : l ≤ l := by
```

Comment:
This should work afaik:
```
theorem implies_refl (l : MagmaLaw α) : l ≤ l :=
  fun _ _ => rfl
```

Reply:
nope, it doesn't:
```
type mismatch
  rfl
has type
  ?m.104 = ?m.104 : Prop
but is expected to have type
  satisfiesPhi x✝ l : Prop
  ```
  but even if it did, what's wrong with the current proof? looks perfectly reasonable to me.

Reply:
Sorry for the misleading comment.